api/v1alpha1/common.go

```diff
@@ -54,6 +58,25 @@ type PulsarMessaging struct {
 	TLSSecret string `json:"tlsSecret,omitempty"`
 }
 
+type PulsarStateStore struct {
+	// The service url points to the state store service
+	// By default, the state store service is bookkeeper table service
+	ServiceURL string `json:"serviceUrl"`
+
+	// The state store config for Java runtime
+	JavaProvider *PulsarStateStoreJavaProvider `json:"javaProvider,omitempty"`
+}
+
+type PulsarStateStoreJavaProvider struct {
+	// The java class name of the state store provider implementation
+	// The class must implement `org.apache.pulsar.functions.instance.state.StateStoreProvider` interface
+	// If not set, `org.apache.pulsar.functions.instance.state.BKStateStoreProviderImpl` will be used
+	ClassName string `json:"className"`
+
+	// The configmap of the configuration for the state store provider
+	Config string `json:"config,omitempty"`
```

Comment:
should use `map[string]string`?

Reply:
use `*Config` instead, since `*Config` supports `map[string]interface{}`